### DISCUSSION TITLE
Weekly Meeting #1

### DISCUSSION THREAD

**Discussion by norswap** [0xFableOrg/0xFable#112] [General]
Agenda:

- Git merge flow (Norswap)
- Lore & art direction for PoC cards: ideas + brainstorm

**Comment by norswap** [comment]
My summary:

- Git merge flow
     - use rebase inside feature branches (PRs)
          - as often as possible: `git fetch master && git rebase master`
          - in particular, make sure this is done before merging into master, so that every commit in the branch is logically on top of the head of master
          - some commits inside feature branches are expected to be non-functional (tests/functionality broken), that's expected and sometimes desirable to make better small commits
     - merge PRs into master, so that the master history from now on is only merge commits from pull requests
         - this allows us to go from working state to working state better, and pinpoint regressions more easily
- reformatting with prettier
     - wait until the two current PRs are merged, then reformat the whole typescript codebase with prettier before carrying on with further work
     - configuration:
        ```
        {
            "trailingComma": "all",
            "tabWidth": 4,
            "semi": false,
            "singleQuote": false
        }
        ```
        (trailing comma: having them makes the diff cleaner)
- lore & art
     - (summarizes the discussion, which includes input from other than the initiator)
     - Norswap
          - let's use a "DnD-style" setting: there are a lot of mysterious things everywhere (mystical creatures, dungeons, objects of power)
              - this means a lot of cards will naturally not be faction-aligned
              - this gives us more creative freedom to design iconic & distinctive cards
          - factions vs resources: in MtG, colors encompass both playstyles and color, but the factions are a bit stereotypical / boring because of this
              - the point of multiple types of resources is gameplay — strategic decision for resource management, including when to go multi-resources and how much
              - the draft mechanism for resources (playing a "colored" card face down to generate one resource of the color) is a lot more forgiving than magic, might encourage going multi-resource more
              - we could split factions between multiple colors, and have tribal effects across both colors and factions (kind of like "tribal" mechanisms in MtG)
     - Jasin
          - old card game he worked on: a card game but where creatures lived on an hex grid and had to move around to attack
               - also a single hero had to be chosen, which was aligned to a faction, and you had to have a minimum amount of cards from that faction
          - let's keep that as a "difficult" use case for developing our extensibility framework
               - in particular, the key thing to avoid is having to rewrite all the cards, so the game state interface that gets passed in to each card's logic should be compatible
          - extra attributes: could be defined as a `cardTypeID → attribute` mapping in a separate contract
                - we could also define a bunch of attributes on each card, not used in the base game but that could be leveraged in extensions
                    - we can't balance those, as we don't know the use case! but the cost is small, at worst people ignore those
     - Aritra
          - we could build Legend of the 5 Rings (LotFR) style stories on top of a faction mechanism
              - e.g. earn points for your faction or for yourself, unlock faction rewards, earn more points if beating an enemy (strong against) your faction, faction war events, etc...
              - this could/should be a different metagame from the base game, for instance by putting restrictions on decks to use a minimum amount of faction-aligned cards
              - this is another good use case that should be able to be built on top of the base game, maybe run by us but "opt-out" for extensions
          - select "blessings" or effects for your deck, similar to planeswalkers in MtG or heroes in Jasin's game
          
I most likely forgot some stuff!

**Reply by eviterin** [reply]
- Norswap had a Bob Ross moment drawing the git rebase/merge flows. 
    - I'm sure there were some smart points but what I remember is that I should spam 'git fetch master' then 'git rebase master'
- Art discussion
    - Norswap 
        - started by bringing up (proposing?) a setting where there are old relics and machinery which kind of can be used to introduce different cards. 
        - magic's mana system. Which is not always very good because sometimes you have a bad draw and you don't have the correct colors needed and you just lose. 
        - 0xFable can have both a mana system and a faction system which could work together. Simple example: some cards can buff factions - some cards can buff colors. 
        - Cards can be placed face-down to generate mana of that card's color. (How? ZK proofs). 
        - Small discussion on game constraints. Traditional card games have constraint that they should be possible to play in real life. 0xFable can disregard that constraint and do _wiiiiild_ shit. 
        - something about MUD and rewriting the game architecture in that?
    - Aritra 
        - continued by preparing us mentally for what he was about to say
        - POINTS - players can gain points by beating other factions. Say player 'A' vs player 'B'. If A wins then they gain points both in their own faction and the opponent's faction. These points can then be used for various things, such as changing faction. This could be a springboard to make 0xFable a place where players can shape their own story. 
        - Both Jasin & Norswap thought this was a cool idea. Norswap mentioned that this type of thing should be something that could be built on-top by anyone.
    - Jasin  
         - said a lot of smart and cool stuff - you had to be there
         - explained about game he had worked on previously where cards also had a movement attribute because the game was played on a board (like chess). 
         - Norswap said that this use-case should be something to have in mind when we design the architecture - making sure it _could_ be done. 

Feel free to correct me.

**Comment by ultraviolet10** [comment]
My ideas/notes: 

On `attack` flow: refer to architecture docs for first draft of the POC, game-play mechanics will evolve over time.

On game lore/setting: 
- Create the idea that everyone can derive their own "fable" and contribute to the stories of their respective factions.
- Cards can use hue gradients from https://0to255.com/ for design clarity. See: [Duel Masters](https://www.youtube.com/watch?v=By7fgOGT97o).
- Players should enter 0xFable for the first time and it should allow them to choose a faction based on a bunch of questions (focused on personality, preferences). A direct reference to this would be discord's [Hypesquad](https://support.discord.com/hc/en-us/articles/360007553672-HypeSquad-House-Breakdown) concept, and God of War's [difficulty](https://images.app.goo.gl/v8RdAgUwCwoeqbU68) guide. We should aim to define houses/factions/regions based on this, and as their gameplay evolves their experiences evolve.
- It would be cool to measure the success of a player against players from another faction. A Scarlet Pirate beating four Black Mages in a row should ideally signify some form of progress within the game, for which we could use the concept of experience _points_ (forgive me @norswap). I do think some form of progress measurement does help with continued engagement. 
- If we're doing weapons, it'd be very cool to have certain creatures matched with certain weapons, so people could use them as boosters - composite strategies, basically.
